plugin/vault.hashicorp.com/v1alpha1/vaultsecrettransformer/VaultSecretTransformer.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/k8sdeps/kunstruct"
+	"sigs.k8s.io/kustomize/api/resid"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/yaml"
+)
+
+type plugin struct {
+	rmf              *resmap.Factory
+	ldr              ifc.Loader
+	secretReceiver   secretReceiver
+	types.TypeMeta   `json:",inline"`
+	types.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+}
+
+//noinspection GoUnusedGlobalVariable
+// nolint: golint
+var KustomizePlugin plugin
+
+func (p *plugin) Config(h *resmap.PluginHelpers, c []byte) error {
+	p.rmf = h.ResmapFactory()
+	p.ldr = h.Loader()
+
+	var err error
+	if p.secretReceiver, err = initVaultClient(); err != nil {
+		return fmt.Errorf("can't initialize Vault client: %s", err)
+	}
+
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *plugin) Transform(m resmap.ResMap) error {
+
+	targetId := resid.NewResId(VaultSecretGvk, "")
+	for _, res := range m.Resources() {
+		if gvkEquals(targetId, resmap.GetOriginalId(res)) {
+			marshaled, err := res.MarshalJSON()
```

Comment:
please make the body of this if into a function, e.g.
```
if gvkEquals(targetId, resmap.GetOriginalId(res)) {
  if err := processResource(m, res); err != nil {
    return err
 }
}
```
in that function, don't log - instead return a wrapped error, e.g.
```
return errors.Wrapf(err, "Can't marshal an object: %s: %s", res, res.GetName())
```
logging for debugging purposes adds too much noise.